tutorials/12-gaussian-process-latent-variable-model/12-gplvm-advi.jl

```diff
@@ -0,0 +1,144 @@
+# Load Turing.
+using Turing
+#  using AbstractGPs, Plots
+using AbstractGPs, Stheno, KernelFunctions, Random, Plots
+
+# Background reading:
+# https://jmlr.org/papers/v6/lawrence05a.html
+# http://proceedings.mlr.press/v9/titsias10a/titsias10a.pdf
+
+#  using Memoization
+#  Turing.setrdcache(true)
+
+# Load other dependencies
+using Distributions, LinearAlgebra
+using VegaLite, DataFrames
+
+using DelimitedFiles
+oil_matrix = readdlm("Data.txt", Float64)
+labels = readdlm("DataLabels.txt", Float64)
+labels = mapslices(x -> findmax(x)[2], labels, dims=2)
+
+# choose AD backend
+#  using Zygote # Tracker supported? check it?
+#  Turing.setadbackend(:zygote)
+#  TODO why does this give an error?
+
+#  SqExponentialKernel is alias of RBFKernel
+sekernel(α, σ) = SqExponentialKernel() ∘ ARDTransform(α)
+
+@model function GPLVM(Y, ndim=4, ::Type{T} = Float64) where {T}
+
+  # Dimensionality of the problem.
+  N, D = size(Y)
+  # dimensions of latent space
+  K = ndim
+  noise = 1e-6
+
+  # Priors
+  α ~ MvLogNormal(MvNormal(K, 1.0))
+  σ ~ MvLogNormal(MvNormal(D, 1.0))
+  # use filldist for Zygote compatibility
+  Z ~ filldist(Normal(0., 1.), K, N)
```

Comment:
The following will fix it:
```suggestion
  Z_vec ~ filldist(Normal(0., 1.), K * N)
  Z = reshape(Z_vec, K, N)
```

It's due to the meanfield approximation constructed not handling higher-dim arrays properly. I have a PR open with a fix for this, but it never got any attention; I'll try to get that PR merged so the above workaround won't be necessary.

Reply:
Great, thanks for the quick response!

Reply:
I am running into some sort of race condition/infinity loop now. It's affecting both MCMC inference and ADVI. For ADVI I get an error message, whereas MCMC just seems to run forever. This happens when I uncomment these 2 lines, I added a reproducible example in the latest commit:
#  using Zygote # Tracker supported? check it?
#  Turing.setadbackend(:zygote)
Should I raise an issue for this?

Reply:
I think so -- I've been running into this on another project too, and honestly I'm kind of glad that someone else can replicate this issue outside of my project.

Reply:
This is probably also what's timing out all the CIs 😕

Reply:
https://github.com/TuringLang/DistributionsAD.jl/issues/180